examples/01-basic/11-file-uploading-uppy/App.tsx

```diff
@@ -27,45 +40,9 @@ export default function App() {
         type: "image",
       },
     ],
-  });
-
-  const uppy = new Uppy().use(XHR, {
-    endpoint: "https://tmpfiles.org/api/v1/upload",
-  });
-
-  uppy.on("upload-success", (file: any, response: any) => {
-    // File should be removed from the uppy instance after successful upload.
-    if (response.status === 200) {
-      uppy.removeFile(file.id);
-    }
-
-    const block = editor.document[2];
-
-    // The file URL is returned in the response body need to be modified to be downloadable.
-    let fileURL = response.body.data.url;
-    fileURL = fileURL.replace("tmpfiles.org/", "tmpfiles.org/dl/");
-
-    const updateData = {
-      props: {
-        name: file?.name,
-        url: fileURL,
-      },
-    };
-    editor.updateBlock(block, updateData);
-  });
-
-  uppy.on("upload-error", (file: any, error: any) => {
-    console.error("Upload error:", error);
+    uploadFile,
   });
 
   // Renders the editor instance using a React component.
-  return (
-    <BlockNoteView editor={editor} filePanel={false}>
-      <FilePanelController
-        filePanel={() => {
-          return <Dashboard uppy={uppy} width={400} height={500} />;
-        }}
-      />
-    </BlockNoteView>
-  );
+  return <BlockNoteView editor={editor} />;
```

Comment:
Why did we remove the Dashboard? This was quite nice and part of the original requirements

Reply:
It is not possible to use dashboard when we need to use the existing uploadFile which is in [UploadTab.tsx](https://github.com/TypeCellOS/BlockNote/blob/main/packages/react/src/components/FilePanel/DefaultTabs/UploadTab.tsx)

Either we can implement in this way [https://github.com/TypeCellOS/BlockNote/pull/842/files/06053acfc998b70845b8bf689e8a8241d5770496..a3ed8fc41d24ec00614e6bd72a83da8fcd0541b6#diff-48f7b0a43a51a764debb003cbd5e52382948064e2af0b1a76f1c2968b8c1d714](https://github.com/TypeCellOS/BlockNote/pull/842/files/06053acfc998b70845b8bf689e8a8241d5770496..a3ed8fc41d24ec00614e6bd72a83da8fcd0541b6#diff-48f7b0a43a51a764debb003cbd5e52382948064e2af0b1a76f1c2968b8c1d714)

Or implement in this way
[https://github.com/TypeCellOS/BlockNote/pull/842/files/56efb1f33d262fcd3ca203b8973bdc84d05c01db#diff-10e76f1f7e0513fa475c80821b41e5a38ca78b76e840007cee48c2e30f3cdda9](https://github.com/TypeCellOS/BlockNote/pull/842/files/56efb1f33d262fcd3ca203b8973bdc84d05c01db#diff-10e76f1f7e0513fa475c80821b41e5a38ca78b76e840007cee48c2e30f3cdda9)

Reply:
We can use the second option, but still implement uploadFile. In that case for the panel, the Dashboard and direct uppy upload is used.

For other cases (such as file drop), the editor will call `uploadFile` and your implementation will make sure files are uploaded via uppy.

Hope this makes sense why I think we need both

Reply:
@YousefED I would suggest to go with 2nd only as it supports both file drop as well as selecting files for uploading. Implementing both on an example wont be possible. If this is fine, I can update my PR with the 2nd implementation

Reply:
fixed, as you can see in my changes it's certainly possible